client/wfprev-war/src/main/angular/src/app/components/edit-project/project-details/fiscal-year-projects/fiscal-year-projects.component.ts

```diff
@@ -0,0 +1,72 @@
+import { CommonModule } from '@angular/common';
+import { Component, OnInit} from '@angular/core';
+import { MatExpansionModule } from '@angular/material/expansion';
+import { ActivatedRoute } from '@angular/router';
+import { ProjectService } from 'src/app/services/project-services';
+import { convertFiscalYear } from 'src/app/utils/tools';
+import { MatTableModule } from '@angular/material/table';
+
+@Component({
+  selector: 'app-fiscal-year-projects',
+  standalone: true,
+  imports: [MatExpansionModule,CommonModule, MatTableModule],
+  templateUrl: './fiscal-year-projects.component.html',
+  styleUrl: './fiscal-year-projects.component.scss'
+})
+export class FiscalYearProjectsComponent implements OnInit{
+  projectFiscals: any[] = [];
+
+  //this fake data will be replaced by fiscalActivities once it is ready
+  activities = [
+    { name: 'Site Prep', description: 'Preparing the site for seeding', startDate:'2024-02-15', endDate: '2024-03-15', completedHectares: 50 },
+    { name: 'Seeding', description: 'Planting seeds for reforestation', endDate: '2024-05-10', completedHectares: 100 },
+    { name: 'Monitoring', description: 'Tracking vegetation growth', endDate: '2024-08-25', completedHectares: 75 },
+  ];
+  displayedColumnsComplete: string[] = ['name', 'description', 'endDate', 'completedHectares'];
+  displayedColumnsPlanned: string[] = ['name', 'description', 'startDate', 'endDate', 'plannedHectares'];
+
+  completeFiscalDisplayedColumns: string[] = ['name', 'description', 'endDate', 'completedHectares'];
+  planFiscalStatusMap: Record<string, string> = {
+    "DRAFT": "Draft",
+    "PROPOSED": "Proposed",
+    "IN_PROG": "In Progress",
+    "COMPLETE": "Complete",
+    "ABANDONED": "Abandoned",
+    "PREPARED": "Prepared"
+  };
+
+  convertFiscalYear = convertFiscalYear
+  constructor(
+    private route: ActivatedRoute,
+    private projectService: ProjectService
+  ) {}
+  ngOnInit(): void {
+      this.loadProjectFiscals();
+  }
+
+  loadProjectFiscals(): void {
```

Comment:
More of a general comment throughout the code here, and in a lot of other PRs I've seen recently...

But in this function, what happens when projectGuid is null? What happens if it fails? What if the observer errors out? Not even a message when it's null in the console?

Lots of use of `?` operators when things might be null/undefined, but no handling on what happens if they actually are?

Basically, when something goes wrong, it will be catastrophic and things in the app will start to lock and behave very poorly or unexpectedly. We need to make sure we're actually handling errors, not just ignoring them.

Reply:
Thanks for the feedback! I will add Error handling for the API calls in this and other component by wrapping the subscription in an error handler to avoid silent failures. And some console errors. Will do this in separate PR

Reply:
Yeah for sure a separate PR. It'll help with application stability and debugging if we're catching and handling error cases vs. just throwing undefined objects around until the screen locks and needs a refresh